haystack/preview/components/preprocessors/text_document_splitter.py

```diff
@@ -0,0 +1,96 @@
+from typing import List, Dict, Any, Literal, Tuple
+
+from more_itertools import windowed
+
+from haystack.preview import component, Document, default_from_dict, default_to_dict
+
+
+@component
+class TextDocumentSplitter:
+    """
+    Splits a list of text documents into a list of text documents with shorter texts.
+    This is useful for splitting documents with long texts that otherwise would not fit into the maximum text length of language models.
+    """
+
+    def __init__(
+        self, split_by: Literal["word", "sentence", "passage"] = "word", split_length: int = 200, split_overlap: int = 0
+    ):
+        """
+        :param split_by: The unit by which the document should be split. Choose from "word" for splitting by " ",
+        "sentence" for splitting by ".", or "passage" for splitting by "\n\n".
+        :param split_length: The maximum number of units in each split.
+        :param split_overlap: The number of units that each split should overlap.
+        """
+
+        self.split_by = split_by
+        if split_by not in ["word", "sentence", "passage"]:
+            raise ValueError("split_by must be one of 'word', 'sentence' or 'passage'.")
+        if split_length <= 0:
+            raise ValueError("split_length must be greater than 0.")
+        self.split_length = split_length
+        if split_overlap < 0:
+            raise ValueError("split_overlap must be greater than or equal to 0.")
+        self.split_overlap = split_overlap
+
+    @component.output_types(documents=List[Document])
+    def run(self, documents: List[Document]):
+        """
+        Splits the documents by split_by after split_length units with an overlap of split_overlap units.
+        Returns a list of documents with the split texts.
+        :param documents: The documents to split.
+        :return: A list of documents with the split texts.
+        """
+
+        if not documents or not isinstance(documents, list) or not isinstance(documents[0], Document):
+            raise TypeError("TextDocumentSplitter expects a List of Documents as input.")
+        split_docs = []
+        for doc in documents:
+            if doc.text is None:
+                raise ValueError("TextDocumentSplitter only works with text documents but document.text is None.")
```

Comment:
Let's specify which doc ID raised this error

```suggestion
                raise ValueError(f"TextDocumentSplitter only works with text documents but document.text for document ID {doc.id} is None.")
```